### PR TITLE
rshift use default >> operator

### LINKED ISSUE #40032: Overflow when binary shifting negative integer tensors stored on CPU
## 🐛 Bug

Overflow when binary shifting integer tensors stored on CPU

## To Reproduce

_in:_ torch.tensor(-2) >> 1
_out:_ tensor(9223372036854775807)

Note that when using cuda or when doing the operation with python integers, the result does not overflow like this and instead performs the expected integer division by a power of 2

_in:_ torch.tensor(-2, device='cuda') >> 1
_out:_ tensor(-1, device='cuda:0')

_in:_ -2 >> 1
_out:_ -1

## Expected behavior

The output of torch.tensor(-2) >> 1 should be -1. I understand that this is probably caused by the 2's complement way of storing signed integers, but the result should still be consistent with the cuda result and the python native result.

## Environment

PyTorch version: 1.5.0
Is debug build: No
CUDA used to build PyTorch: 10.2

OS: Microsoft Windows 10 Famille
GCC version: Could not collect
CMake version: Could not collect

Python version: 3.7
Is CUDA available: Yes
CUDA runtime version: Could not collect
GPU models and configuration: Could not collect
Nvidia driver version: Could not collect
cuDNN version: Could not collect

Versions of relevant libraries:
[pip] numpy==1.18.1
[pip] numpydoc==0.9.2
[pip] torch==1.5.0
[pip] torchvision==0.6.0
[conda] blas                      1.0                         mkl
[conda] cudatoolkit               10.2.89              h74a9793_1
[conda] mkl                       2020.0                      166
[conda] mkl-service               2.3.0            py37hb782905_0
[conda] mkl_fft                   1.0.15           py37h14836fe_0
[conda] mkl_random                1.1.0            py37h675688f_0
[conda] numpy                     1.18.1           py37h93ca92e_0
[conda] numpy-base                1.18.1           py37hc3f5095_1
[conda] numpydoc                  0.9.2                      py_0
[conda] pytorch                   1.5.0           py3.7_cuda102_cudnn7_0    pytorch
[conda] torchvision               0.6.0                py37_cu102    pytorch

CPU: Intel(R) Core(TM) i7-8750H CPU @ 2.20GHz
64 bits system, x64 cpu

cc @VitalyFedyunin @ngimel @mruberry @ezyang @SsnL @gchanan @zou3519

### PR DESCRIPTION
**Pull request by xwang233**
Fix https://github.com/pytorch/pytorch/issues/40032
Also see https://github.com/pytorch/pytorch/pull/35339

### CODE REVIEW

File: test/test_torch.py [hunk 1]
```diff
@@ -6698,7 +6698,7 @@ def test_signed_shift(self, device, dtype):
         a = torch.tensor([-10, 10], device=device, dtype=dtype)  # [11...1110110, 1010]
         expected_l = torch.tensor([-40, 40], device=device, dtype=dtype)  # [11...11011000, 101000]
         self.assertEqual(a << 2, expected_l)
-        expected_r = torch.tensor([torch.iinfo(dtype).max - 4, 5], device=device, dtype=dtype)  # [1111...111011, 101]
+        expected_r = torch.tensor([-5, 5], device=device, dtype=dtype)  # [1111...111011, 101]
```
Comment (ngimel) [inline test/test_torch.py:6701]: can you also add `compare_with_numpy` test (look in test_torch for example usage) to make sure that our behavior agrees with numpy?

File: test/test_torch.py [hunk 2]
```diff
@@ -6691,15 +6691,19 @@ def test_inverse(self, device):
...
     @onlyOnCPUAndCUDA
     @dtypes(torch.int8, torch.int16, torch.int32, torch.int64)
     def test_signed_shift(self, device, dtype):
         "Ensure that signed integer bit shifting works as expected."
-        a = torch.tensor([-10, 10], device=device, dtype=dtype)  # [11...1110110, 1010]
+        values = [-10, 10]
+        a = torch.tensor(values, device=device, dtype=dtype)  # [11...1110110, 1010]
         expected_l = torch.tensor([-40, 40], device=device, dtype=dtype)  # [11...11011000, 101000]
         self.assertEqual(a << 2, expected_l)
+        self.compare_with_numpy(lambda x: x << 2, lambda x: np.left_shift(x, 2), values, device, dtype)
```
Comment (ngimel) [inline test/test_torch.py:6703]: you can send `a` directly as `tensor_like` argument to `compare_with_numpy`.

Comment (xwang233) [reply test/test_torch.py]: Fixed. I forgot that I can remove device and dtype in kwargs.

**Comment by xwang233** [comment]
@ngimel Is there any update on this?

**Review by ngimel — APPROVED** [review]

**Pull request closed without merging**